jobs/kola-aws.Jenkinsfile

```diff
@@ -0,0 +1,62 @@
+@Library('github.com/coreos/coreos-ci-lib') _
+
+def streams
+node {
+    checkout scm
+    streams = load("streams.groovy")
+    pod = readFile(file: "manifests/pod.yaml")
+}
+
+properties([
+    pipelineTriggers([]),
+    parameters([
+      choice(name: 'STREAM',
+             // list devel first so that it's the default choice
+             choices: (streams.development + streams.production + streams.mechanical),
+             description: 'Fedora CoreOS stream to test'),
+      string(name: 'VERSION',
+             description: 'Fedora CoreOS Build ID to test',
+             defaultValue: '',
+             trim: true),
+      string(name: 'S3_STREAM_DIR',
+             description: 'Fedora CoreOS S3 Stream Directory',
+             defaultValue: '',
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override coreos-assembler image to use',
+             defaultValue: "coreos-assembler:master",
+             trim: true)
+    ])
+])
+
+currentBuild.description = "[${params.STREAM}] - ${params.VERSION}"
+
+def s3_stream_dir = params.S3_STREAM_DIR
+if (s3_stream_dir == "") {
+    s3_stream_dir = "fcos-builds/prod/streams/${params.STREAM}"
+}
+
+cosaPod(image: params.COREOS_ASSEMBLER_IMAGE, memory: "256Mi",
+        secrets: ["aws-fcos-builds-bot-config", "aws-fcos-kola-bot-config"]) {
+
+    def ami, ami_region
+    stage('Fetch Metadata') {
+        shwrap("""
+        export AWS_CONFIG_FILE=/run/kubernetes/secrets/aws-fcos-builds-bot-config/config
+        cosa init --branch ${params.STREAM} https://github.com/coreos/fedora-coreos-config
+        cosa buildprep --ostree --build=${params.VERSION} s3://${s3_stream_dir}/builds
+        """)
+
+        def basearch = shwrapCapture("cosa basearch")
+        def meta = readJSON file: "builds/${params.VERSION}/${basearch}/meta.json"
+        if (meta.amis.size() > 0) {
+            ami = meta['amis'][0]['hvm']
+            ami_region = meta['amis'][0]['name']
+        } else {
+            throw new Exception("No AMI found in metadata for ${params.VERSION}")
+        }
+    }
+
+    fcosKola(cosaDir: env.WORKSPACE, parallel: 5,
+             extraArgs: "-p aws --build ${params.VERSION} --aws-ami ${ami} --aws-region ${ami_region}")
```

Comment:
how are the credentials getting passed in here?

previously we had `export AWS_CONFIG_FILE=\${AWS_FCOS_KOLA_BOT_CONFIG}` right before we called `cosa kola run`.

For GCP we pass in `--gce-json-key=` so that's taken care of.

Reply:
Yup, good catch!